### ISSUE TITLE
Scroll Overwiew - full description and other View

### ISSUE THREAD

**Issue by SleepingShadow-bit** [DinoDevs/GladiatusCrazyAddon#298]
Adjust the scrolls overview so that you have 4 tables.
Prefix and suffix learned.
Prefix and suffix not learned. And those that you have not learned to display as plain text but "grayed out"

Keep the link to gladiatus-tools.com
![Overview](https://user-images.githubusercontent.com/8767665/146074250-1e9a13be-98d8-4a54-be49-d9900f7d086a.png)
![Screenshot_3](https://user-images.githubusercontent.com/8767665/146074276-f94a9552-bf73-4f5a-bd79-03eb110b2e00.png)

**Comment by GramThanos** [comment]
The problem is that in order to do that, we have to generate a list with the scrolls names for each language, which we don't have.

Ok I will think about it and maybe ask the list from Skarsburning.

**Comment by Zitronenen** [comment]
@GramThanos @FrutyX 
The wish is not a must!
If it is not possible, then close - as always a nice to have :)

**Comment by FrutyX** [comment]
Nah, I just wish it was easier to do, we see.

**Comment by FrutyX** [comment]
> The problem is that in order to do that, we have to generate a list with the scrolls names for each language, which we don't have.


Wait a minute, https://gladiatus-tools.com/ has prefixes and suffixes in different languages, you just change the "COMMUNITY" and the scroll names will change, just try to put it on Czech Republic, you'll see it's gonna change.

**Comment by GramThanos** [comment]
I now, but first we have to get permission from Skarsburning (it's his website).

**Comment by FrutyX** [comment]
Skars is known now as Michalus? I didn't know that, there is no connection between https://gladiatus.gamerz-bg.com/ and https://gladiatus-tools.com/ or at least I don't see it.

**Comment by GramThanos** [comment]
Hmmm no, I think you are right and I mixed the names. Michalus is the admin that created the page.

**Comment by FrutyX** [comment]
Yep, now it makes sense, so you need Michalus.

You have to ask for it first then... or you can snatch and grab... but that's not your style. What? It's just a scroll list :D.

**Comment by SleepingShadow-bit** [comment]
I don't know if @GreatApo  has already asked Michalus. But I have the prefix overview of all languages from Michalu's side (copy and paste) here - maybe for testing?!.
However, I do not want to share this here if Apo or @GramThanos does not want it.
Let me know

**Comment by FrutyX** [comment]
> The problem is that in order to do that, we have to generate a list with the scrolls names for each language, which we don't have.


By the way, since the last update, you can change your interface language, which changes the names of the scrolls as well, might be helpful?!

**Comment by SleepingShadow-bit** [comment]
Today i saw the Scroll Code... How does the share code work with the scrolls?
![Screenshot_2023-12-12-23-24-54-246_com microsoft rdc androidx-edit](https://github.com/DinoDevs/GladiatusCrazyAddon/assets/8767665/e1604fd7-2167-4d57-9248-157fdd753596)

**Comment by GreatApo** [comment]
> Today i saw the Scroll Code... How does the share code work with the scrolls?
> ![Screenshot_2023-12-12-23-24-54-246_com microsoft rdc androidx-edit](https://github.com/DinoDevs/GladiatusCrazyAddon/assets/8767665/e1604fd7-2167-4d57-9248-157fdd753596)
> 

It doesn't :P

I was experimenting with compressing the known or unknown scrolls. This part works, but I never scripted the decompress - show another player's scrolls...

**Comment by SleepingShadow-bit** [comment]
Oooh maybe u will did it

**Comment by FrutyX** [comment]
So, yeah, it took me a while, maybe you noticed https://github.com/DinoDevs/GladiatusCrazyAddon/blob/master/source/core/source/gca.data.recipeNames.js, this currently isn't connected in any way with the addon and the current scroll implementation, but.. we have it.

So if anyone is feeling fancy, it can be used for this or other GCA features.

**Comment by FrutyX** [comment]
Hello!

This Issue has been selected to be merged into our Feature Requests & Enhancements overview Issue -  #530.

Please note that this does not change anything, the request is still taken in account, but it's moved for better and cleaner GitHub management.

Thank you for your feedback and ideas! We appreciate them :).

**Comment by GreatApo** [comment]
> What I still don't like is that tables don't have headers labels ("#", "Name", "Level", "Details") - but the tables are not wide enough to fit them. 

You can replace the **level** header with the levels icon and the **details** with the magnifying glass.

**Comment by GramThanos** [comment]
@warkadiusz the `gca_tools.ajax.cached.known_scrolls()` is used on all the pages that want to check the known scrolls (guild storage, market?)

**Comment by FrutyX** [comment]
Alright, ignore the previous message, it just needed a kick.

So I checked my other account, that has no scrolls learned and I see this:

![Image](https://github.com/user-attachments/assets/1e13fa65-57a0-4c23-8cbf-9501454219b0)

Also, the links are now broken too.

This is probably gonna need more testing for various cases, I can check probably later, nowt now.

**Comment by FrutyX** [comment]
Google Chrome

And yes, I always download the source folder from your branch, I can see the file has changed the structure, it is changed. The problem is somewhere else.

**Comment by FrutyX** [comment]
A quick analysis tells me that **lang="null"**

**Comment by FrutyX** [comment]
@warkadiusz This is probably on me, but please, if you have the time, can you check that all languages work?

Because in recipeNames, Czech is "cz", however, GCA has "cs", so of course, that gives me the undefined again...

I made the translated list from the game (option value=""), not from GCA, which is why there may be some differences.

Thank you.

**Comment by FrutyX** [comment]
Random feedback: Instead of showing (U) we could also show something like"💀".

Matter of opinion tho.

**Comment by FrutyX** [comment]
> Thanks a lot!
> ...

1. Having fallbacks is always a good idea (Thanos should do them more often, heheheheh)
2. Nice work checking out the languages
3. I am not sure, I had them separately because you see how long it is, and I wasn't sure how we will need to edit the whole file. It was also a bit cleaner this way, but... I can't give you an answer, I don't think it's needed tho.
4. Right align looks good to me!
5.  You still need to check what is happening with the links to Gladiatus-tools, I am getting `https://s51-en.gladiatus.gameforge.com/game/1,1-1,0` instead of expected  `https://en.gladiatus-tools.com/equipment?item=1,1-1,0`

**Comment by GramThanos** [comment]
@FrutyX (-.-) The versions are starting with a 4 and this hasn't changed for 7 years? Because this was the last time I rewrote the whole addon structure... to make it fast, stable and with fallbacks.

**Comment by FrutyX** [comment]
You can put this: `filter: drop-shadow(0 0 1px black);` on the skull, if you want, just visual.

Other than that, I am pretty satisfied with what you have done. I have nothing to add.

I will do some testing again once it's fully merged anyway.

**Comment by FrutyX** [comment]
Two things:

1. Make sure all text you have added can be translated.
2. Change line 1953 to:
```
var container = document.getElementById("content_2nd");
for (var i = 0, row; row = container.getElementsByTagName("tr")[i]; i++) {
	if (i === 0 || i === 164) continue;
		row.style.color = (array[i]) ? 'red' : 'green';
}
```

See what I did there?

**Comment by GramThanos** [comment]
> Two things:
> 
> 1. Make sure all text you have added can be translated.
> 2. Change line 1953 to:
> ```
> var container = document.getElementById("content_2nd");
> for (var i = 0, row; row = container.getElementsByTagName("tr")[i]; i++) {
> 	if (i === 0 || i === 164) continue;
> 	row.style.color = (array[i]) ? 'red' : 'green';
> }
> ```
> 
> See what I did there?
> 
> 

Optimised:

```
let rows = document.getElementById("content_2nd").getElementsByTagName("tr");
for (let i = rows.length - 1; i >= 0; i--) {
 	if (i === 0 || i === 164) continue;
 	rows[i].style.color = (array[i]) ? 'red' : 'green';
}
```

Improvements:
- Access the getElementsByTagName only 1 time, instead of on each loop
- use let to release the i variable when the loop ends
- inverse the loop to have a faster check

**Comment by FrutyX** [comment]
> Optimised:

Thanks for the optimization!

(That's Apo's code btw, I just added the skip :D)

**Comment by GreatApo** [comment]
> > Optimised:
> 
> Thanks for the optimization!
> 
> (That's Apo's code btw, I just added the skip :D)
> 
> 
> 

Thanos spent more energy writing his post than the speed gains that he introduced... But if the loop was millions of lines, that is the best way to do it.

**Comment by GramThanos** [comment]
> 
> (That's Apo's code btw, I just added the skip :D)
> 

Oups... I hope he wrote it years ago :P

**Comment by FrutyX** [comment]
Yes, getting little off-topic here, but boy, that response time

![Image](https://github.com/user-attachments/assets/c09eb1bf-034d-466b-bbef-3f02ff3d157f)

![Image](https://github.com/user-attachments/assets/b1424954-1675-467b-8c21-d5ba04c00c56)

😆

**Comment by GramThanos** [comment]
> Thanos spent more energy writing his post than the speed gains that he introduced...

I was in the subway checking my email on the phone :P... thus it was efficient management of my energy.

> But if the loop was millions of lines, that is the best way to do it.

Tell that to the poor duo core trying to run this loop 100 times, crying painfully through its cpu circles while waiting his DDR2 ram to recover all the lines that his poor CACH missed... its a tragedy 😭

**Comment by FrutyX** [comment]
@warkadiusz Watch and learn. There is nothing better for GCA, than some little brother rivarly 😄

**Comment by GramThanos** [comment]
> @warkadiusz Watch and learn. There is nothing better for GCA, than some little brother rivarly 😄 

Haha, no rivarly, just poking each other.

**Comment by FrutyX** [comment]
> Haha, no rivarly, just poking each other.

Oh, I see. Then we will have to push much harder. Turn this into a gladiator arena and maybe we'll even see GCA v5 :D.

Alright, enough fun.

@warkadiusz Do what you need to do and you can open a PR.

I say it again, well done, I am glad we used the scroll translations in the end, and that it's just not sitting there.
